p2p/src/authenticated/actors/dialer.rs

```diff
@@ -149,10 +149,7 @@ impl<
             self.dial_peers(&mut tracker, &mut supervisor).await;
 
             // Sleep for a random amount of time up to the dial frequency
-            let wait = Duration::from_millis(
-                self.runtime
-                    .gen_range(0..self.dial_frequency.as_millis() as u64),
-            );
+            let wait = self.runtime.gen_range(Duration::default()..self.dial_frequency);
```

Comment:
I think this should be okay based on what copilot says, I'm not overly familiar with range in rust

Reply:
https://github.com/rust-random/rand/issues/82

Reply:
This looks correct to me 👍